modules/frontend/pipeline/collector_http.go

```diff
@@ -35,29 +39,79 @@ func (r httpCollector) RoundTrip(req *http.Request) (*http.Response, error) {
 		return nil, err
 	}
 
+	err = addNextAsync(ctx, r.consumers, resps, r.combiner, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return r.combiner.HTTPFinal()
+}
+
+func addNextAsync(ctx context.Context, consumers int, resps Responses[combiner.PipelineResponse], c combiner.Combiner, callback func() error) error {
+	respChan := make(chan combiner.PipelineResponse)
+	overallErr := atomic.Error{}
+	wg := sync.WaitGroup{}
+
+	setErr := func(err error) {
+		overallErr.CompareAndSwap(nil, err)
+	}
+
+	if consumers <= 0 {
+		consumers = 10
+	}
```

Comment:
Unsure about this, I understand 0 as "sync".

Reply:
i consider `consumers == 1` sync and `consumers <= 0` as simply invalid which is why i chose a default. 

technically `consumers == 1` is synchronous just with silly channel shenanigans.